DDRec/include/DDRec/DCH_info.h

```diff
@@ -0,0 +1,348 @@
+    DCH_layer nlayersPerSuperlayer = {0};
+    /// input number of superlayers
+    /// superlayer is an abstract level of grouping layers used to
+    /// parametrize the increment of cells in each layer
+    DCH_layer nsuperlayers = {0};
+    /// Calculated as dch_nlayersPerSuperlayer * dch_nsuperlayers
+    DCH_layer nlayers = {0};
+
+    /// global twist angle
+    /// alternating layers will change its sign
+    DCH_angle_t  twist_angle = {0};
+
+    /// Cell width for the first layer
+    double first_width = {0};
+    /// Cell radius for the first layer
+    DCH_length_t first_sense_r = {0};
+
+    void Set_lhalf(DCH_length_t _dch_Lhalf){Lhalf=_dch_Lhalf;}
+    void Set_rin  (DCH_length_t _dch_rin  ){rin  = _dch_rin; }
+    void Set_rout (DCH_length_t _dch_rout ){rout = _dch_rout;}
+
+    void Set_guard_rin_at_z0  (DCH_length_t _dch_rin_z0_guard  ){guard_inner_r_at_z0  = _dch_rin_z0_guard;   }
+    void Set_guard_rout_at_zL2(DCH_length_t _dch_rout_zL2_guard){guard_outer_r_at_zL2 = _dch_rout_zL2_guard; }
+
+    void Set_ncell0              (int _ncell0              ){ncell0               = _ncell0;              }
+    void Set_ncell_increment     (int _ncell_increment     ){ncell_increment      = _ncell_increment;     }
+
+    void Set_nlayersPerSuperlayer(int _nlayersPerSuperlayer){nlayersPerSuperlayer = _nlayersPerSuperlayer;}
+    void Set_nsuperlayers        (int _nsuperlayers        ){nsuperlayers         = _nsuperlayers;        }
+
+    void Set_ncell_per_sector(int _ncell_per_sector){ncell_per_sector = _ncell_per_sector;}
+
+    void Set_twist_angle (DCH_length_t _dch_twist_angle ){twist_angle = _dch_twist_angle;}
+
+    void Set_first_width  (double _first_width  ){first_width   = _first_width;   }
+    void Set_first_sense_r(double _first_sense_r){first_sense_r = _first_sense_r; }
+
+
+    /// Get number of cells in a given layer
+    ///   ncells = number of wires/2
+    int Get_ncells(int ilayer){return database.at(ilayer).nwires/2;}
+
+    /// Get phi width for the twisted tube and the step (phi distance between cells)
+    DCH_angle_t Get_phi_width(int ilayer){return (TMath::TwoPi()/Get_ncells(ilayer))*dd4hep::rad;}
+
+    /// phi positioning, adding offset for odd ilayers
+    /// there is a staggering in phi for alternating layers, 0.25*cell_phi_width*(ilayer%2);
+    DCH_angle_t Get_cell_phi_angle(int ilayer, int nphi){ return (Get_phi_width(ilayer) * (nphi + 0.25*(ilayer%2)));}
+
+    /// calculate superlayer for a given ilayer.
+    /// WARNING: division of integers on purpose!
+    int Get_nsuperlayer_minus_1(int ilayer){ return int((ilayer-1)/nlayersPerSuperlayer);}
+
+    /// Calculate radius at z=L/2 given at z=0
+    DCH_length_t Radius_zLhalf(DCH_length_t r_z0) const {
+        return r_z0/cos(twist_angle/2/dd4hep::rad);
+    }
+
+    /// tan(stereoangle) = R(z=0)   / (L/2) * tan( twist_angle/2)
+    DCH_angle_t stereoangle_z0(DCH_length_t r_z0) const {
+        return atan( r_z0/Lhalf*tan(twist_angle/2));
```

Comment:
```suggestion
        return atan( r_z0/Lhalf*tan(twist_angle/2/dd4hep::rad));
```
You `/dd4hep::rad` in the `cos` above, so for consistency? Also in the next function for `sin` ?